File: LearnAppServerAPI/LearnAppServerAPI/Data/Entities/FlashcardsSet.cs
namespace LearnAppServerAPI.Data.Entities
{
    public class FlashcardsSet
    {
        public int Id { get; set; }
        public User Editor { get; set; }
        public int EditorId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool IsPublic { get; set; }
        public DateTime Date { get; set; }
        public List<Flashcard> Flashcards { get; set; }
    }
}

File: mobileApp/lib/data/flashcards_set.dart
import '../data/flashcard.dart';

class FlashcardsSet {
  int Id = 0;
  int EditorId = 0;
  String Name = '';
  String Description = '';
  String Date = '';
  List<Flashcard> Flashcards = [];

  FlashcardsSet.empty();

  FlashcardsSet(this.Id, this.EditorId, this.Name, this.Description, this.Date,
      this.Flashcards);

  FlashcardsSet.fromJson(Map<String, dynamic> flashcardsSetMap) {
    Id = flashcardsSetMap['id'] ?? 0;
    EditorId = flashcardsSetMap['editorId'] ?? 0;
    Name = flashcardsSetMap['name'] ?? '';
    Description = flashcardsSetMap['description'] ?? '';
    Date = flashcardsSetMap['date'] ?? '';
    Flashcards = (flashcardsSetMap['flashcards'] as List).map((i) {
      return Flashcard.fromJson(i);
    }).toList();
  }

  Map<String, dynamic> toJson() => {
        'editorId': EditorId,
        'name': Name,
        'description': Description,
        'date': Date,
      };
}
